src/protocol/prss.rs

```diff
@@ -533,6 +568,42 @@ pub mod test {
         assert_eq!(r3_l, r2_r);
     }
 
+    #[test]
+    fn three_party_bitwise_fields() {
+        fn b2_to_b10(a: &[Fp2]) -> u128 {
+            let mut v = 0;
+            #[allow(clippy::cast_possible_truncation)]
+            for (i, &b) in a.iter().enumerate() {
+                v += 2_u128.pow(i as u32) * b.as_u128();
+            }
+            v
+        }
+
+        const IDX: u128 = 7;
+        let (p1, p2, p3) = make_participants();
+        let step = UniqueStepId::default();
+
+        // These tests do not check that left != right because
+        // the field might not be large enough.
+        let (rb1_l, rb1_r) = p1
+            .indexed(&step)
+            .generate_bitwise_fields::<Fp31, Fp2, u128>(IDX);
+        let (r1_l, r1_r) = (b2_to_b10(&rb1_l), b2_to_b10(&rb1_r));
+        let (rb2_l, rb2_r) = p2
+            .indexed(&step)
+            .generate_bitwise_fields::<Fp31, Fp2, u128>(IDX);
+        let (r2_l, r2_r) = (b2_to_b10(&rb2_l), b2_to_b10(&rb2_r));
+        let (rb3_l, rb3_r) = p3
+            .indexed(&step)
+            .generate_bitwise_fields::<Fp31, Fp2, u128>(IDX);
+        let (r3_l, r3_r) = (b2_to_b10(&rb3_l), b2_to_b10(&rb3_r));
+
+        // Verify for each party i, l_i == r_(i-1)
+        assert_eq!(r1_l, r3_r);
+        assert_eq!(r2_l, r1_r);
+        assert_eq!(r3_l, r2_r);
```

Comment:
You're just checking that this is a valid replicated secret sharing - but we should also validate that $a ∈ {0, 1}$